pkg/operator/ceph/object/notification/controller.go

```diff
@@ -0,0 +1,258 @@
+func Add(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context, opConfig opcontroller.OperatorConfig) error {
+	if err := addNotificationReconciler(mgr, &ReconcileNotifications{
+		client:           mgr.GetClient(),
+		context:          context,
+		opManagerContext: opManagerContext,
+	}); err != nil {
+		return err
+	}
+
+	return addOBCLabelReconciler(mgr, &ReconcileOBCLabels{
+		client:           mgr.GetClient(),
+		context:          context,
+		opManagerContext: opManagerContext,
+	})
+}
+
+func addNotificationReconciler(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Info("successfully started")
+
+	// Watch for changes on the OBC CRD object
+	err = c.Watch(&source.Kind{Type: &cephv1.CephBucketNotification{}}, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephBucketNotification object and makes changes based on the state read
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNotifications) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileNotifications) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// fetch the CephBucketNotification instance
+	notification := &cephv1.CephBucketNotification{}
+	err := r.client.Get(r.opManagerContext, request.NamespacedName, notification)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("CephBucketNotification %q resource not found. Ignoring since resource must be deleted.", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketNotification %q", request.NamespacedName)
+	}
+
+	// get the topic associated with the notification
+	bucketTopic := &cephv1.CephBucketTopic{}
+	topicName := types.NamespacedName{Namespace: notification.Namespace, Name: notification.Spec.Topic}
+	if err := r.client.Get(r.opManagerContext, topicName, bucketTopic); err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Infof("CephBucketTopic %q not provisioned yet", topicName)
+			return waitForRequeueIfTopicNotReady, nil
+		}
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketTopic %q", topicName)
+	}
```

Comment:
I think there is a very subtle error in terminology here that could be confusing to users. In reality, it's not that the topic hasn't been provisioned: the topic resource has not even been created by the user. It's possible that the user has mis-configured the notification to refer to a nonexistent topic. 

This is similar to k8s Pods when they reference a volume that doesn't exist. Pods will register an event on the pod and retry periodically.
```
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    4m4s                 default-scheduler  Successfully assigned default/test-dep-6598d6c4d-58kq5 to minikube
  Warning  FailedMount  2m1s                 kubelet            Unable to attach or mount volumes: unmounted volumes=[from-secret], unattached volumes=[from-secret kube-api-access-kj62z]: timed out waiting for the condition
  Warning  FailedMount  116s (x9 over 4m4s)  kubelet            MountVolume.SetUp failed for volume "from-secret" : secret "nonexistent-secret" not found
```

I think we should do a similar thing from this case as well. There are mechanisms in place to do this without writing a lot of code. See https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/controller.go, for example.